website/content/en/docs/best-practices/multi-tenancy.md

```diff
@@ -0,0 +1,162 @@
+in order to respond to changes in their clusters’ needs, such as by resizing control plane nodes to provide more resources.
+
+**Following are some general recommendations to manage the resources:**
+
+- **MUST** declare resource requests for both CPU and memory.
+- **OPTIONALLY** setting the resources limit for Memory and CPU (Note that if your project is integrated with OLM then, admins would be able to customize it via the [subscription][olm-subscriptions]). Also, be aware that our project was built with Kubebuilder or SDK scaffold then for the `Operator/manager`(See `config/manager/manager.yaml`), then some values which allow you to get started are scaffolded by default, however, you ought to optimize based on the needs.
+- **Should** provide the mechanisms for [Monitoring compute & memory resource usage][k8s-metrics]. Kubebuilder and SDK provide some [metrics][kb-metrics] which can help you with. Note that cluster admins can use it to monitor and resize the Operator and its Operands.
+- **OPTIONALLY** ensuring that the Operators uses [liveness health check][k8s-probes] which can be used to check the `CPU` performance. If your project was built with Kubebuilder or SDK scaffold then, it is scaffolded by default for the `Operator/manager`. (See `config/manager/manager.yaml`).
+- **Should** allow admins to customize the `requests/limits` amounts defined for the Pod/Deployment Operands created by the Operator and not keep them fixed on the code.
+- **Should** document how cluster admins can customize/rightsize the resources limits for the Operator and Operands.
+
+You also might consider providing solutions that allow vertical pod autoscaler to manage arbitrary scaling for your Operators. [More info][k8s-vertical-pod-scaling]. 
+And then, you also might look for allowing [horizontal pod autoscaling][k8s-autoscaling] for the Operands. In this case, see that resource requests are also needed for horizontal pod autoscaling to work.
+
+**NOTE:** If you are using the k8s OpenShift vendor you might want to check the doc [Automatically adjust pod resource levels with the vertical pod autoscaler
+][ocp-pod-scaling].
+
+### But, what happens
+
+#### when the resource requests are not set?
+
+**If not set:** 
+- configurations made by the cluster administrators such as [ResourceQuota][k8s-resources-quotas] might not work. 
+- the Operators consumers might face resource shortage on a node when resource usage later increases, for example, during a daily peak in request rate.
+- the Operators consumers might install the Operator in a node/namespace which does not have the minimal resources available.
+
+#### when the resource limits are not set?
+
+**If not set:** wrong configurations or code implementations can consume all resources available, affecting other components on cluster. 
+Also, it might let the containers more vulnerable such as to Dos Attacks. [More info][docker-security].
```

Comment:
@fgiloux, 
> If no CPU limit is set. Nothing bad will neither happen to the node where the pod is running nor the cluster
That is **NOT** true. Please, check the Kubernetes docs. I added here: https://github.com/operator-framework/operator-sdk/pull/5353#discussion_r753102148

Also, see:

- If you do not specify a CPU limit: https://kubernetes.io/docs/tasks/configure-pod-container/assign-cpu-resource/#if-you-do-not-specify-a-cpu-limit
- Motivation for CPU requests and limits: https://kubernetes.io/docs/tasks/configure-pod-container/assign-cpu-resource/#motivation-for-cpu-requests-and-limits

Regards ALL other points please see that they are already addressed in this document. Read `#### when the resource limits exceed?`

Reply:
Sorry but I respectfully disagree. 
From the link you shared: "The Container has no upper bound on the CPU resources it can use. The Container could use all of the CPU resources available on the Node where it is running."
This is not that terrible thing that you may think because:
- It won't be able to do that if there is CPU resource contention on the node
- After every process has received what it requested, free cycles are allocated by the  Completely Fair Scheduler according to the request set (not the limit). 
"
    The Pod can have bursts of activity where it makes use of CPU resources that happen to be available.
    The amount of CPU resources a Pod can use during a burst is limited to some reasonable amount.
"
Except for edge cases like performance testing or pods that request a very big amount of CPU compared to other workload I don't see a benefit of limiting the CPU that can be used (not considering bills of cloud providers or other things that should be detected by a monitoring system)

Reply:
Yep

Reply:
We chat and I think we could come up with a text that sorted it out.